[CI] CircuitBreakingOperationsTests testDeterminizeWithLimitedBreakerProducesCorrectResult failing

**Build Scans:**
- [elasticsearch-intake #37755 / part1](https://gradle-enterprise.elastic.co/s/4ogj4t3vqo4vi)
- [elasticsearch-periodic-platform-support #12105 / debian-12_platform-support-unix](https://gradle-enterprise.elastic.co/s/qf5gscho557om)
- [elasticsearch-pull-request #137851 / part-1](https://gradle-enterprise.elastic.co/s/pnlekqlx3tm5u)
- [elasticsearch-periodic-platform-support #12100 / sles-12_platform-support-unix](https://gradle-enterprise.elastic.co/s/2igpgcittiafo)
- [elasticsearch-periodic-platform-support #12100 / opensuse-leap-15_platform-support-unix](https://gradle-enterprise.elastic.co/s/dwpbfx56fuizo)
- [elasticsearch-periodic #12123 / adoptopenjdk17_checkpart1_java-fips-140-3-matrix](https://gradle-enterprise.elastic.co/s/v4hczswcx2ihs)
- [elasticsearch-periodic #12123 / adoptopenjdk17_checkpart1_java-fips-140-3-matrix](https://gradle-enterprise.elastic.co/s/yvlfvuqxk5gy4)
- [elasticsearch-periodic-platform-support #12080 / sles-12_platform-support-unix](https://gradle-enterprise.elastic.co/s/dy3npqodqv772)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.lucene.util.automaton.CircuitBreakingOperationsTests.testDeterminizeWithLimitedBreakerProducesCorrectResult" -Dtests.seed=C27C79F995E83AFD -Dtests.locale=st-LS -Dtests.timezone=Asia/Saigon -Druntime.java=26
```

**Applicable branches:**
8.19

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.lucene.util.automaton.CircuitBreakingOperationsTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testDeterminizeWithLimitedBreakerProducesCorrectResult'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: null
```

**Issue Reasons:**
- [8.19] 1 consecutive failures in test testDeterminizeWithLimitedBreakerProducesCorrectResult
- [8.19] 1 consecutive failures in step part1
- [8.19] 1 consecutive failures in pipeline elasticsearch-intake
- [8.19] 8 failures in test testDeterminizeWithLimitedBreakerProducesCorrectResult (3.1% fail rate in 261 executions)
- [8.19] 2 failures in step sles-12_platform-support-unix (25.0% fail rate in 8 executions)
- [8.19] 2 failures in step adoptopenjdk17_checkpart1_java-fips-140-3-matrix (33.3% fail rate in 6 executions)
- [8.19] 3 failures in pipeline elasticsearch-periodic-platform-support (50.0% fail rate in 6 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/server/src/test/java/org/elasticsearch/common/lucene/search/AutomatonQueriesTests.java
+++ b/server/src/test/java/org/elasticsearch/common/lucene/search/AutomatonQueriesTests.java
@@ -171,7 +171,9 @@ public class AutomatonQueriesTests extends ESTestCase {
         if (a1 == a2) {
             return true;
         }
-        return subsetOf(a2, a1) && subsetOf(a1, a2);
+        Automaton normalizedA1 = Operations.removeDeadStates(a1);
+        Automaton normalizedA2 = Operations.removeDeadStates(a2);
+        return subsetOf(normalizedA2, normalizedA1) && subsetOf(normalizedA1, normalizedA2);
     }
 
     private void assertCollapsedAndInvalidRegexHandled(String pattern, String expectedCollapsed) {

--- a/server/src/test/java/org/elasticsearch/lucene/util/automaton/CircuitBreakingOperationsTests.java
+++ b/server/src/test/java/org/elasticsearch/lucene/util/automaton/CircuitBreakingOperationsTests.java
@@ -126,9 +126,11 @@ public class CircuitBreakingOperationsTests extends ESTestCase {
     }
 
     private static void assertAcceptSameStrings(Automaton expected, Automaton actual) {
+        Automaton normalizedExpected = Operations.removeDeadStates(expected);
+        Automaton normalizedActual = Operations.removeDeadStates(actual);
         assertTrue(
             "Circuit-breaking determinize should produce the same language as Lucene's determinize",
-            Operations.subsetOf(expected, actual) && Operations.subsetOf(actual, expected)
+            Operations.subsetOf(normalizedExpected, normalizedActual) && Operations.subsetOf(normalizedActual, normalizedExpected)
         );
     }
 }

